### PR TITLE
[GPU] Added more exception handlings

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Added exception handling for internal failure of program build and reshape for MAX_BATCH_SIZE

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp [hunk 1]
```diff
@@ -16,6 +16,8 @@ namespace ov {
 namespace runtime {
 namespace intel_gpu {
 
+const uint32_t default_num_streams_for_tput = 2;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp:19]: @myshevts  I did same thing with your PR here

Comment (myshevts) [reply src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp:19]: sure, again, this is  a plaholder so [when someone implement the GetMetric for the #command-streamers, this var indicates the place where to call that, as we discussed with@vladimir-paramuzov](https://github.com/openvinotoolkit/openvino/pull/7883/commits/b52768c2cc9dbc0aa4ec7c4999de215e3a066b67#diff-a9b4426da3bf04732eec03004bd4351c68573a79f54646420b24189ef1f73e7bR18)

**Review by myshevts — APPROVED** [review]

File: src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp [hunk 2]
```diff
@@ -47,6 +48,9 @@ struct Config {
         adjustKeyMapValues();
     }
 
+    uint32_t GetDefaultNStreamsForThroughputMode() const {
+        return 2;
+    }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp:53]: @myshevts @vladimir-paramuzov Replaced this as a function in Config. Please review this!

Comment (myshevts) [reply src/plugins/intel_gpu/include/intel_gpu/plugin/device_config.hpp:53]: I'm fine, but again, should this be a metric... maybe we can reuse the METRIC_KEY(RANGE_FOR_STREAMS)?
 the "higher" va;ue of that should match #CSSs... @vladimir-paramuzov whta do you think?

**Pull request merged**